Analytics/Analytics/Network/HttpClient.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  HttpClient.swift
+//  Analytics
+//
+//  Created by Satheesh Kannan on 24/09/24.
+//
+
+import Foundation
+
+// MARK: - HttpClientRequests
+/**
+ This protocol is designed to execute predefined network requests.
+ */
+protocol HttpClientRequests {
+    func getConfiguarationData()
+    func postBatchEvents(_ batch: String)
+}
+
+// MARK: - HttpClient
+/**
+ This class provides the implementation for the `HttpClientRequests` protocol.
+ */
+final class HttpClient {
+    let analytics: AnalyticsClient
+    
+    init(analytics: AnalyticsClient) {
+        self.analytics = analytics
+    }
+    
+    private func prepareGenericUrlRequest(for requestType: HttpClientRequestType) -> URLRequest? {
+        guard let url = self.prepareRequestUrl(for: requestType) else { return nil }
+        var urlRequest = URLRequest(url:url)
+        urlRequest.httpMethod = requestType.httpMethod
+        urlRequest.allHTTPHeaderFields = requestType.headers(analytics)
+        return urlRequest
+    }
+    
+    private func prepareRequestUrl(for requestType: HttpClientRequestType) -> URL? {
+        guard var url = URL(string: requestType.url(analytics).trimmedUrlString) else { return nil }
+        url = url.appending(path: requestType.endpoint)
+        
+        if requestType == .configuration {
+            url = url.appendQueryParameters(Constants.configQueryParams)
+        }
+        return url
+    }
+    
+    private func runRequset(_ request: URLRequest) {
+        HttpNetwork.perform(request: request) { result in
+            switch result {
+            case .success(let response):
+                print(response.toJSONString ?? "Bad response")
+            case .failure(let error):
+                print(error.localizedDescription)
+            }
+        }
+    }
+}
+
+// MARK: - HttpClientRequests
+
+extension HttpClient: HttpClientRequests {
+    func getConfiguarationData() {
+        guard let urlRequest = self.prepareGenericUrlRequest(for: .configuration) else { return }
+        self.runRequset(urlRequest)
+    }
+    
+    func postBatchEvents(_ batch: String) {
+        guard var urlRequest = self.prepareGenericUrlRequest(for: .events) else { return }
+        urlRequest.httpBody = batch.utf8Data
+        
+        self.runRequset(urlRequest)
+    }
+}
+
+// MARK: - HttpClientRequestType
+/**
+ An enume that provides all network request-related data based on the request type.
+ */
+enum HttpClientRequestType {
+    case configuration
+    case events
+    
+    func url(_ analytics: AnalyticsClient) -> String {
+        return switch self {
+        case .configuration: analytics.configuration.controlPlaneUrl
+        case .events: analytics.configuration.dataPlaneUrl
+        }
+    }
+    
+    var endpoint: String {
+        return switch self {
+        case .configuration: "sourceConfig"
```

Comment:
Please keep it like this:
```suggestion
        case .configuration: "/sourceConfig"
```
In Android, we are following the pattern of adding `/` in the endpoint.

Reply:
No need to add `/` as a prefix if you are appending a path component to the URL object.

Reply:
I used the below function call to append the url & endpoint.
Reference: https://developer.apple.com/documentation/foundation/url/3988449-appending

```
let urlString: String = "https://www.apple.com"

let endpoint = "test"

let url = URL(string: urlString).appending(path: endpoint)
```